### PR TITLE
Compiler errors and warnings are reported by PMD reporter

### PR DESCRIPTION
**Pull request by Uranium-92**
Closes https://github.com/oclint/oclint/issues/398.

### CODE REVIEW

File: oclint-reporters/reporters/PMDReporter.cpp [hunk 1]
```diff
@@ -25,27 +25,62 @@ class PMDReporter : public Reporter
...
+
+    void writeFooter(std::ostream &out)
+    {
+        out << "</pmd>";
+    }
+
+    void writeCompilerError(std::ostream &out, const Violation &compilerError)
+    {
+        auto ruleDescription = PMDRuleDescription("compiler error",
+                                                  "compiler",
```
Comment (ryuichis) [inline oclint-reporters/reporters/PMDReporter.cpp]: Wondering if it makes sense to put this, `compiler warning`, and `clang static analyzer` all under the `clang` ruleset.

**Comment by ryuichis** [comment]
I started to worry the compiler errors and warnings for certain projects might be too noisy and that our tool output gets sunk into the noises. Do you think it's a valid concern, and make this only enabled by an additional flag when invoke the tool? We can make all other reporters work the same way.

File: oclint-reporters/reporters/PMDReporter.cpp (hunk 1 above)
Comment (Uranium-92) [inline oclint-reporters/reporters/PMDReporter.cpp]: Seems reasonable to me. Thanks! Will fix it today.

**Comment by Uranium-92** [comment]
Good point about noise in reports.

But what I think:
1. if the project we analyze is compiled with clang, clang errors mean absolute necessity to fix them;
2. if the project we analyze is compiled with clang, clang warnings will appear in build log of the project and there seems to be no big need to duplicate these warnings in analyzer reports (but if I had a choice I would prefer to get all issues in one place - in analyzer report, even if it is a duplication; if some warnings should be ignored we can explicitly suppress them in code and they will disappear from both build log and analyzer report);
3. if project is compiled with the compiler other than clang, clang errors or warnings may indicate serious issues in analyzed code;
4. OCLint requires source file to be compilable by clang to be able to analyze the file, so if clang can not compile the file, OCLint can say nothing about presence or absence of violations in this file - in this situation it seems reasonable to me to note this in report (this point is not applicable to warnings though).

To sum up:
IMO, it is reasonable to make reporting of warnings optional (by adding special command-line flag for this), but errors reporting should be required (maybe it is worth adding phrase "please be aware that this error will prevent OCLint from analyzing this source code" in error description message).

Anyway adding new command-line flag and handling of it in all reporters seems to be the subject of separate Pull request.
